subcommand/lifecycle-sidecar/command_test.go

```diff
@@ -61,27 +68,164 @@ func TestRun_ServiceConfigFileMissing(t *testing.T) {
 	require.Contains(t, ui.ErrorWriter.String(), "-service-config file \"/does/not/exist\" not found")
 }
 
-func TestRun_ConsulBinaryMissing(t *testing.T) {
+func TestRun_FlagValidation_ConsulBinaryMissing(t *testing.T) {
 	t.Parallel()
 	ui := cli.NewMockUi()
 	cmd := Command{
 		UI: ui,
 	}
 
 	// Create a temporary service registration file
+	tmpDir, configFile := createServicesTmpFile(t)
+	defer os.RemoveAll(tmpDir)
+
+	configFlag := "-service-config=" + configFile
+
+	responseCode := cmd.Run([]string{configFlag, "-consul-binary=/not/a/valid/path"})
+	require.Equal(t, 1, responseCode, ui.ErrorWriter.String())
+	require.Contains(t, ui.ErrorWriter.String(), "-consul-binary \"/not/a/valid/path\" not found")
+}
+
+func TestRun_FlagValidation_InvalidLogLevel(t *testing.T) {
+	t.Parallel()
+	tmpDir, configFile := createServicesTmpFile(t)
+	defer os.RemoveAll(tmpDir)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+	responseCode := cmd.Run([]string{"-service-config", configFile, "-consul-binary=consul", "-log-level=foo"})
+	require.Equal(t, 1, responseCode, ui.ErrorWriter.String())
+	require.Contains(t, ui.ErrorWriter.String(), "unknown log level: foo")
+}
+
+// Test that we register the services.
+func TestRun_ServicesRegistration(t *testing.T) {
+	t.Parallel()
+	tmpDir, configFile := createServicesTmpFile(t)
+	defer os.RemoveAll(tmpDir)
+
+	a, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
```

Comment:
For my own education, I'd like to know the trade-offs between using `NewTestServerConfigT` vs. the other test server setups in the sdk. Also, is the datacenter necessary? I would have thought the defaults would be sufficient.

Reply:
There are 2 functions for setting up the server that we want to use:
- `NewTestServerT` - to use when you want to use just the default values
- `NewTestServerConfigT` - to use when you need to tweak some config values; it takes a callback that allows you to overwrite config values as I did in this test.

There is also the `NewTestServerConfig`, but I don't think we need to use it because it doesn't take `t`.

Yeah, you're right. I don't think we need the primary dc setting in this case.